tasks/galera.yaml

```
@@ -0,0 +1,86 @@
+---
+- name: Check if grastate.dat exists
+  stat:
+    path: /var/lib/mysql/grastate.dat
+  register: grastate
+
+- name: Check if cluster is save to bootstrap
+  when: grastate.stat.exists
+  slurp:
+    path: /var/lib/mysql/grastate.dat
+  register: bootstrap
+
+- name: Check if mariadb is already runnning
+  stat:
+    path: /var/run/mysqld/mysqld.pid
+  register: pid
+
+- name: Fail if cluster cannot be initialzed
+  fail:
+    msg: "cluster initializer is not ready to bootstrap cluster"
+  when:
+    - mariadb.galera.initializer == True
+    - bootstrap.content is defined
+    - "'Safe to Bootstrap: 0' in bootstrap.content | b64decode"
+    - pid.stat.exists == False
+
+- block:
+
+  - name: stop mariadb
+    service:
+      name: mariadb
+      state: stopped
+
+  - name: remove wsrep_ready
+    file:
+      path: /tmp/wsrep_ready
+      state: absent
+
+  - name: repeatedly write cluster state to wsrep_ready
+    cron:
+      name: "check cluster"
+      minute: "*"
+      job: "echo \"SHOW STATUS LIKE 'wsrep_ready';\" | mysql -u root | grep 'wsrep_ready' > /tmp/wsrep_ready"
+
+  - name: Initialize cluster
+    when: mariadb.galera.initializer == True
+    command: /usr/bin/galera_new_cluster
+
+  - name: Wait that cluster is initialized
+    when: mariadb.galera.initializer == True
+    wait_for:
+      path: /tmp/wsrep_ready
+      search_regex: "wsrep_ready\tON"
```

Together with the other tasks this is not idempotent. Can't we just use a shell task instead that exits once the cluster is ready? This way we would only need a shell task instead of multiple cron and file tasks.

Something like this:

```suggestion
    changed_when: false
    shell: |
      while true; do
        if echo "SHOW STATUS LIKE 'wsrep_ready';" | mysql | grep -E 'wsrep_ready\tON'; then
          break
        fi
      done
```

Also is the `-u root` required? The task is already running as root.

---

Now I see your review comment in the other merge request and that you were already planning to do that. https://github.com/punktDe/proserver-ansible-mariadb/pull/2/files#r441632675

---

yep

---

The problem here is, that I want the task to fail after a certain time, if the node does not become ready. I already asked @lyxys for advice on that, and he agreed to help.

---

Fixed in 0fe09cb9c332c49e811cad6b2069f210a697f62e